src/lib/es2015.iterable.d.ts

```diff
@@ -8,23 +8,30 @@ interface SymbolConstructor {
     readonly iterator: symbol;
 }
 
-interface IteratorResult<T> {
-    done: boolean;
-    value: T;
+interface IteratorYieldResult<TYield> {
+    done: false;

```

Comment:
`done` could be omitted in this case.
 ```suggestion
    done?: false;
```

Reply:
Unlike my response to https://github.com/microsoft/TypeScript/pull/30790/files/1cc57b8c3f9ab25f077737515cb3cb86fa8089b3#r282494422, this may be feasible. I will test to verify whether this will cause any problems in the type system.